### PR TITLE
(FACT-2562) Correctly load custom and external fact directories

### PR DESCRIPTION
**Pull request by BogdanIrimie**

### CODE REVIEW

File: spec/facter/facter_spec.rb [hunk 1]
```diff
@@ -286,7 +286,7 @@
 
   describe '#search_path' do
     it 'sends call to LegacyFacter' do
```
Comment (glennsarti) [inline spec/facter/facter_spec.rb:288]: Test description is now wrong

File: spec/facter/facter_spec.rb [hunk 2]
```diff
@@ -302,7 +302,7 @@
 
   describe '#search_external_path' do
     it 'sends call to LegacyFacter' do
```
Comment (glennsarti) [inline spec/facter/facter_spec.rb:304]: Test description is now wrong

File: spec/facter/facter_spec.rb (hunk 1 above)
Comment (IrimieBogdan) [reply spec/facter/facter_spec.rb]: Updated the test description

File: spec/facter/facter_spec.rb (hunk 2 above)
Comment (IrimieBogdan) [reply spec/facter/facter_spec.rb]: Updated the test description

**Review by sebastian-miclea — APPROVED** [review]

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**